cognite_toolkit/_cdf_tk/loaders/_resource_loaders/transformation_loaders.py

```diff
@@ -377,6 +426,32 @@ def delete(self, ids: SequenceNotStr[str | int]) -> int:
             self.client.transformations.delete(id=existing, ignore_unknown_ids=True)
         return len(existing)
 
+    def _update_nonce(self, items: Sequence[TransformationWrite]) -> None:
+        for item in items:
+            if not item.external_id:
+                raise ToolkitRequiredValueError("Transformation must have external_id set.")
+            if read_credentials := self._authentication_by_id_operation.get((item.external_id, "read")):
+                item.source_nonce = self._create_nonce(read_credentials)
+            if write_credentials := self._authentication_by_id_operation.get((item.external_id, "write")):
+                item.destination_nonce = self._create_nonce(write_credentials)
+
+    def _create_nonce(self, credentials: OidcCredentials | ClientCredentials) -> NonceCredentials:
+        key = calculate_secure_hash(credentials.dump(), shorten=True)
+        if key in self._nonce_cache:
```

Comment:
nothing is ever added to this cache, so either it needs to be removed (probably bad wrt session count limit) or created nonce-s need to be put in